packages/gatsby/src/bootstrap/load-plugins/validate.js

```diff
@@ -21,89 +32,102 @@ const getBadExports = (plugin, pluginAPIKeys, apis) => {
   return badExports
 }
 
-const getBadExportsMessage = (badExports, exportType, apis) => {
-  const { stripIndent } = require(`common-tags`)
-  const stringSimiliarity = require(`string-similarity`)
+const getErrorContext = (badExports, exportType, currentAPIs, latestAPIs) => {
   let capitalized = `${exportType[0].toUpperCase()}${exportType.slice(1)}`
   if (capitalized === `Ssr`) capitalized = `SSR`
 
-  let message = `\n`
-  message += stripIndent`
-    Your plugins must export known APIs from their gatsby-${exportType}.js.
-    The following exports aren't APIs. Perhaps you made a typo or your plugin is outdated?
-
-    See https://www.gatsbyjs.org/docs/${exportType}-apis/ for the list of Gatsby ${capitalized} APIs
-  `
-
-  badExports.forEach(bady => {
-    message += `\n\n`
-    const similarities = stringSimiliarity.findBestMatch(bady.exportName, apis)
-    const isDefaultPlugin = bady.pluginName == `default-site-plugin`
-    const badExportsMigrationMap = {
```

Comment:
I just removed these. Is there a reason to keep these around in Gatsby 2.x?

Reply:
Ideally we don't remove those. It's definitely less likely that any user will hit that at this point in time (almost year after gatsby@2). Was this causing some headaches with refactoring?

Reply:
Also in screenshot from PR description you had migration tips for those - was this just mockup?

Reply:
Nope! That one (and maybe the other) get covered by the string distance algorithm!